File: week-03/day-2/src/Pirates/Pirate.java
package Pirates;

public class Pirate {
    boolean dead = false;
    static int pirateCounter;
    String name;
    String rank = "Ordinary";
    int consumedRum;
    boolean passedOut = false;

    public Pirate() {
        pirateCounter++;
        this.name = "Pirate" + pirateCounter;
    }

    public void drinkSomeRum() {
        consumedRum += (1 + Math.random()) * 10;
    }


    public void howsItGoingMate() {
        if (dead) {
            System.out.println("He`s dead!");
        } else if (consumedRum > 0) {
            System.out.println("Pour me anudder! " + "Pour me anudder! " + "Pour me anudder! " + "Pour me anudder! ");
        } else {
            System.out.println("Arghh, I'ma Pirate. How d'ya d'ink its goin?");
            passedOut = true;
        }
    }

    public void die() {
        dead = true;
    }

    public void promotion() {
        rank = "Captain";
    }

    public void brawl(Pirate challengedPirate) {
        if (dead) {
            System.out.println("He`s dead!");
        } else {
            double resultOfBrawl = Math.random();
            if (resultOfBrawl < 0.33) {
                die();
            } else if (resultOfBrawl < 0.66) {
                challengedPirate.dead = true;
            } else {
                die();
                challengedPirate.dead = true;
            }
        }
    }

    @Override
    public String toString() {
        return (name + ", Rank: " + rank + ", Consumed rum: " + consumedRum + ", State (passed out/dead): " + passedOut + "/" + dead);
    }

}

File: week-00 (project phase)/Python/Pirates/Pirate.py
from random import randint

class Pirate:
    def __init__(self, name):
        self.name = name
        self.consumed_rum = 0
        self.is_captain = False
        self.alive = True
        self.passed_out = False

    def drink_some_rum(self):
        self.consumed_rum += randint(1,10)

    def howsItGoingMate(self):
        if self.alive == False:
            print("He`s dead!")
        elif self.consumed_rum > 0:
            print("Pour me anudder! " * 4)
        else:
            print("Arghh, I'ma Pirate. How d'ya d'ink its goin?")
            self.passed_out = True

    def die(self):
        self.alive = False

    def promotion(self):
        self.is_captain = True

    def brawl (self, chalange_Pirate):
        if self.alive == False:
            print("He`s dead!")
        else:
            result_of_battle = randint(1,3)
            if result_of_battle == 1:
                self.die()
            elif result_of_battle == 2:
                chalange_Pirate.die()
            else:
                self.die()
                chalange_Pirate.die()

    def __str__(self):
        return "Name: " + self.name + ", consumed rum: " + str(self.consumed_rum)
